### PR TITLE
Update deps + Fix Form Types

### PR DESCRIPTION
**Pull request by erixweb**
# Changes

## API changes

- No changes on the API were made.

## JSDoc/Declarations changes

- I've changed FormType.js to FormType.d.ts and fixed the types in the Form.js and ServerFormRequestPacket.js files

## Behavior changes

- No behaviour changes were made

## Other changes

- Also updated ESLint from 8.46.0 to 8.47.0

### CODE REVIEW

**Comment by kotinash** [comment]
forms are already fixed on this branch: https://github.com/GreenFrogMCBE/GreenFrogMCBE/tree/next

**Comment by kotinash** [comment]
image

**Review by vp817 — APPROVED** [review]

File: src/forms/Form.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /* eslint-disable no-unused-vars */
+// @ts-check
```
Comment (kotinash) [inline src/forms/Form.js:2]: remove this

Comment (vp817) [reply src/forms/Form.js:2]: remove it urself

Comment (kotinash) [reply src/forms/Form.js:2]: no 😡

Comment (vp817) [reply src/forms/Form.js:2]: yes

Comment (kotinash) [reply src/forms/Form.js]: no !

Comment (vp817) [reply src/forms/Form.js]: yes

**Review by vp817 — APPROVED** [review]

File: src/forms/types/FormType.d.ts [hunk 2]
```diff
@@ -13,11 +13,8 @@
...
  */
-module.exports = /** @type {const} */ ({
-	/* A form type, that has support only buttons and text (PLEASE SEE THE EXAMPLE PLUGIN BECAUSE MOST FUNCTIONS WILL NOT WORK FOR THIS TYPE OF FORM: https://github.com/greenfrogmc/ExamplePlugin/blob/29d2675aefccb1c4efaf375ac1aca9895d4913b1/exampleplugin.js#L129) */
-	FORM: "form",
-	/* A form type, that supports every possible field in Minecraft (DOCS: https://github.com/greenfrogmc/ExamplePlugin/blob/29d2675aefccb1c4efaf375ac1aca9895d4913b1/exampleplugin.js#L129) */
-	CUSTOM_FORM: "custom_form",
-	/* A form type, that supports only 2 buttons and can be used for yes/no answers */
-	MODAL_FORM: "modal",
-});
+export const enum FormType {
```
[the remark this answers is not in the document]
Comment (kotinash) [reply src/forms/types/FormType.d.ts:16]: https://github.com/GreenFrogMCBE/GreenFrogMCBE/blob/next/index.d.ts#L781

**Review by kotinash — CHANGES_REQUESTED** [review]

File: src/forms/types/FormType.d.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (vp817) [reply src/forms/types/FormType.d.ts:16]: no it looks good

[the remark this answers is not in the document]
Comment (kotinash) [reply src/forms/types/FormType.d.ts:16]: 💀

**Comment by erixweb** [comment]
Sorry, didn't realise Forms were already fixed. Should I close the PR?

**Comment by kotinash** [comment]
the "update dependencies" part of this pull request is still relevant

**Comment by erixweb** [comment]
The types were also fixed in Form.js. Should I also revert that? They should be correct now though.

**Comment by kotinash** [comment]
> Should I also revert that?

**Pull request merged**